File: webrtc/src/main/java/com/bandwidth/webrtc/RTCBandwidth.java
package com.bandwidth.webrtc;

import com.bandwidth.webrtc.listeners.OnConnectListener;
import com.bandwidth.webrtc.listeners.OnPublishListener;
import com.bandwidth.webrtc.signaling.ConnectionException;

import java.net.URI;

public interface RTCBandwidth {
    void connect(URI uri, OnConnectListener onConnectListener) throws ConnectionException;
    void disconnect();
    void publish(String alias, OnPublishListener onPublishListener);
}

File: webrtc/src/main/java/com/bandwidth/webrtc/RTCBandwidth.kt
package com.bandwidth.webrtc

import com.bandwidth.webrtc.listeners.*
import com.bandwidth.webrtc.signaling.ConnectionException

interface RTCBandwidth {
    @Throws(ConnectionException::class)
    fun connect(deviceToken: String?, onConnectListener: OnConnectListener)

    @Throws(ConnectionException::class)
    fun connect(webSocketUrl: String?, deviceToken: String?, onConnectListener: OnConnectListener)
    fun disconnect()
    fun publish(alias: String, onPublishListener: OnPublishListener?)
    fun unpublish(streamIds: List<String>, onUnpublishListener: OnUnpublishListener)
    fun setOnStreamAvailableListener(onStreamAvailableListener: OnStreamAvailableListener?)
    fun setOnStreamUnavailableListener(onStreamUnavailableListener: OnStreamUnavailableListener?)
}